### ISSUE TITLE
Typer should mark omittable fields for variable `@include` and `@skip`

### ISSUE THREAD

**Issue by potatosalad** [msolomon/argo#8]
_TL;DR:_ Using `@include(if: $var)` or `@skip(if: $var)` should result in a field being marked as "omittable".

For example, a minimal query like this:

```graphql
query($v:Boolean!){__typename@include(if:$v)}
```

Currently, the `argo-js` typer (and the Argo 1.1 spec) result in a WireType that looks like this:

```
{
  data: {
    __typename: STRING<String>
  }?
  errors?: DESC[]?
}
```

However, since the value of `$v` is indeterminate when the WireType is derived, it's expected that the `__typename` field would be marked as omittable like this:

```
{
  data: {
    __typename?: STRING<String>
  }?
  errors?: DESC[]?
}
```

This applies for both the `@include` and `@skip` directives on selections for `Field`, `FragmentSread`, and `InlineFragment` as shown below.

### Service Document

```graphql
schema {
  query: Query
}

interface Child {
  optional: Child
  required: Child!
}

type Object implements Child {
  optional: Child
  required: Child!
}

type Query {
  root: Child
}
```

### Executable Document

```graphql
query FieldOmittable($include: Boolean = false, $skip: Boolean = true) {
  root {
    includeAlways: __typename @include(if: true)
    includeNever: __typename @include(if: false)
    includeVariable: __typename @include(if: $include)
    skipAlways: __typename @skip(if: true)
    skipNever: __typename @skip(if: false)
    skipVariable: __typename @skip(if: $skip)
  }
}
```

### Current WireType

```
{
  data: {
    root: {
      includeAlways: STRING<String>
      includeVariable: STRING<String>
      skipNever: STRING<String>
      skipVariable: STRING<String>
    }?
  }?
  errors?: DESC[]?
}
```

### Expected WireType

```
{
  data: {
    root: {
      includeAlways: STRING<String>
      includeVariable?: STRING<String>
      skipNever: STRING<String>
      skipVariable?: STRING<String>
    }?
  }?
  errors?: DESC[]?
}
```

```diff
--- current
+++ expected
@@ -2,9 +2,9 @@
   data: {
     root: {
       includeAlways: STRING<String>
-      includeVariable: STRING<String>
+      includeVariable?: STRING<String>
       skipNever: STRING<String>
-      skipVariable: STRING<String>
+      skipVariable?: STRING<String>
     }?
   }?
   errors?: DESC[]?
```

Examples for `FragmentSpread` and `InlineFragment` can be seen in [this file](https://github.com/WhatsApp/erlang-argo/blob/main/apps/argo_test/test/argo_typer_SUITE_data/executable_document.graphql) used in the test suite for [`erlang-argo`](https://github.com/WhatsApp/erlang-argo).

**Comment by msolomon** [comment]
Thanks for finding this issue and the detailed report! The spec and reference implementation are now updated to fix it.
